xds/internal/balancer/edsbalancer/eds_test.go

```diff
@@ -323,6 +344,9 @@ func (s) TestConfigChildPolicyUpdate(t *testing.T) {
 	if err := edsLB.waitForChildPolicy(ctx, lbCfgB); err != nil {
 		t.Fatal(err)
 	}
+	if err := edsLB.waitForCounterUpdate(ctx, testServiceName); err != nil {
```

Comment:
1. I think it would be better to have a separate test for `CounterUpdate`. Make a copy of `TestClientWatchEDS` (because it's the simplest), and `waitForCounterUpdate` after the first `UpdateClientConnState`. You can delete the other checks because they are irrelevant.

2. We can keep the checks you added in `TestConfigChildPolicyUpdate`. Add a comment before this line saying that "Counter is updated even though the service name didn't change. The eds_impl will compare the service names, and skip if it didn't change".

Reply:
Made both of these changes.